src/vconsole/vconsole-setup.c

```diff
@@ -54,6 +54,55 @@ static bool is_vconsole(int fd) {
         return ioctl(fd, TIOCLINUX, data) >= 0;
 }
 
+static bool is_allocated(int fd, int idx) {
+        struct vt_stat vcs = {};
+        char vcname[strlen("/dev/vcs") + DECIMAL_STR_MAX(int)];
+
+        if (fd >= 0) {
+                if (ioctl(fd, VT_GETSTATE, &vcs) < 0) {
+                        (void) log_warning_errno(errno, "VT_GETSTATE failed: %m");
+                        return false;
+                }
+                idx = vcs.v_active;
+        }
+
+        xsprintf(vcname, "/dev/vcs%i", idx);
+        if (access(vcname, F_OK) < 0)
+                return false;
+        return true;
+}
```

Comment:
Hmm, so I always prefer if functions return errors properly, and that the caller then decides whether he wants to ignore error conditions. Hence, I think this function (and is_settable()) really should return a negative errno if something fails, as our functions usually do. On top of that it may return 0 or 1, depending on what the actual test result was. It would then be up to the caller to ignore the errors, and figure out whether it wants to treat errors more like 0 or more like 1.